src/QSymbolicsBase/basic_ops_homogeneous.jl

```diff
@@ -82,7 +87,73 @@ Base.show(io::IO, x::SAddOperator) = print(io, "("*join(map(string, arguments(x)
 const SAddBra = SAdd{AbstractBra}
 Base.show(io::IO, x::SAddBra) = print(io, "("*join(map(string, arguments(x)),"+")::String*")") # type assert to help inference
 
-"""Tensor product of quantum objects (kets, operators, or bras)."""
+# defines commutativity for addition of quantum objects
+const CommQObj = Union{SAddBra, SAddKet, SAddOperator} 
+function _in(x::SymQObj, sadd::CommQObj)
+    for i in arguments(sadd)
+        if isequal(x, i)
+            return true
+        end
+    end
+    false
+end
+function Base.isequal(x::CommQObj, y::CommQObj)
+    if typeof(x)==typeof(y)
+        if isexpr(x)
+            if operation(x)==operation(y)
+                ax,ay = arguments(x),arguments(y)
+                (length(ax) == length(ay)) && all(x -> _in(x, y), ax)
+            else
+                false
+            end
+        else
+            propsequal(x,y) # this is unholy
+        end
+    else
+        false
+    end
+end
+
+"""Symbolic application of operator on operator
+
+```jldoctest
+julia> A = SOperator(:A, SpinBasis(1//2)); B = SOperator(:B, SpinBasis(1//2));
+
+julia> A*B 
+AB
+```
+"""
+@withmetadata struct SApplyOp <: Symbolic{AbstractOperator}
```

Comment:
I know this is following the naming convention from `ApplyKet` but here it does not seem to be as easy to follow. Maybe we can remove it to something like `SOpProduct` or something else easier to understand at a glance.

Reply:
or `SMulOperator` to follow the convention from `SAddOperator`